python/tskit/trees.py

```diff
@@ -6515,8 +6515,11 @@ def samples(self, population=None, *, population_id=None, time=None):
         of values of the form `(min_time, max_time)`, only return sample IDs
         whose node time `t` is in this interval such that `min_time <= t < max_time`.
 
```

Comment:
Maybe a longer explanation of exactly how the 'dict' option works, like:
"
The `population` can be specified either by an integer (in which case this is the population ID)
or a dictionary matching information in the population metadata. If it is a dictionary,
it should contain key-value pair(s) that match the metadata of the desired population;
for instance, ``population={'name': 'abc'}`` will select the population that has a 'name' of 'abc'
in metadata. There should be exactly one population that has matching key-value pair(s);
if not, an error is raised.
"